content/articles/php-sessions-and-cookies-to-create-a-secure-remember-me-section-in-a-login-script/Index.md

```diff
@@ -0,0 +1,356 @@
+        $query = "INSERT INTO tbl_token_auth (username, password_hash, selector_hash, expiry_date) values (?, ?, ?,?)";
+        $result = $db_handle->insert($query, 'ssss', array($username, $random_password_hash, $random_selector_hash, $expiry_date));
+        return $result;
+    }
+
+    function update($query) {
+        mysqli_query($this->conn,$query);
+    }
+}
+?>
+```
+#### DataBase controller.php
+```php
+<?php
+class DBController {
+	private $host = "localhost";
+	private $user = "root";
+	private $password = "test";
+	private $database = "db_auth";
+	private $conn;
+    function __construct() {
+        $this->conn = $this->connectDB();
+	}
+	function connectDB() {
+		$conn = mysqli_connect($this->host,$this->user,$this->password,$this->database);
+		return $conn;
+	}
+    function runBaseQuery($query) {
+                $result = mysqli_query($this->conn,$query);
+                while($row=mysqli_fetch_assoc($result)) {
+                $resultset[] = $row;
+                }
+                if(!empty($resultset))
+                return $resultset;
+    }
+
+    function runQuery($query, $param_type, $param_value_array) {
+
+        $sql = $this->conn->prepare($query);
+        $this->bindQueryParams($sql, $param_type, $param_value_array);
+        $sql->execute();
+        $result = $sql->get_result();
+
+        if ($result->num_rows > 0) {
+            while($row = $result->fetch_assoc()) {
+                $resultset[] = $row;
+            }
+        }
+
+        if(!empty($resultset)) {
+            return $resultset;
+        }
+    }
+
+    function bindQueryParams($sql, $param_type, $param_value_array) {
+        $param_value_reference[] = & $param_type;
+        for($i=0; $i<count($param_value_array); $i++) {
+            $param_value_reference[] = & $param_value_array[$i];
+        }
+        call_user_func_array(array(
+            $sql,
+            'bind_param'
+        ), $param_value_reference);
+    }
+
+    function insert($query, $param_type, $param_value_array) {
+        $sql = $this->conn->prepare($query);
+        $this->bindQueryParams($sql, $param_type, $param_value_array);
+        $sql->execute();
+    }
+
+    function update($query, $param_type, $param_value_array) {
+        $sql = $this->conn->prepare($query);
+        $this->bindQueryParams($sql, $param_type, $param_value_array);
+        $sql->execute();
+    }
+}
+?>
+```
+### Conclusion
+If you check the "Remember Me" box, your browser will save a cookie so that if you exit the site window without signing out, you will be immediately logged back in the next time you visit. Otherwise, this will not work unless you set your browser to remember cookies.
+If you select the "Remember Me" option, your browser will save a cookie so that if you close the site window without signing out, you will be automatically logged in the next time you visit. Otherwise, unless you set your browser to remember cookies, this will not work.

```

Comment:
Conclusion should give us a brief overview of what you covered in the article. So this paragraph should be to moved somewhere else.